cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -152,6 +161,19 @@ public synchronized void create(final NamespaceMeta metadata) throws Exception {
     if (!(Principal.SYSTEM.equals(principal) && NamespaceId.DEFAULT.equals(namespace))) {
       authorizationEnforcer.enforce(instanceId, principal, Action.ADMIN);
       authorizer.grant(namespace, principal, ImmutableSet.of(Action.ALL));
+      // Also grant the namespace user all privileges on the namespace, if kerberos is enabled
+      if (kerberosEnabled) {
+        ImpersonationInfo impersonationInfo = impersonationUserResolver.getImpersonationInfo(metadata);
+        String namespacePrincipal = impersonationInfo.getPrincipal();
+        Preconditions.checkNotNull(
```

Comment:
If impersonationInfo is not set at the namespace level, it defaults to ``cConf.get(Constants.Security.CFG_CDAP_MASTER_KRB_PRINCIPAL);``. The default is empty. So this check is wrong - it should check for empty (or just not check at all and allow a runtime failure if it is found to be empty. Will remove.